hadoop-hdds/rocks-native/src/test/java/org/apache/hadoop/hdds/utils/db/managed/TestManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,243 @@
+              new ManagedSSTDumpIterator<ManagedSSTDumpIterator.KeyValue>(tool,
+                  file.getAbsolutePath(), options) {
+
+            @Override
+            protected KeyValue getTransformedValue(Optional<KeyValue> value) {
+              return value.orElse(null);
+            }
+          }) {
+        while (iterator.hasNext()) {
+          ManagedSSTDumpIterator.KeyValue r = iterator.next();
+          Pair<String, Integer> recordKey = Pair.of(new String(r.getKey(),
+              StandardCharsets.UTF_8), r.getType());
+          Assertions.assertTrue(keys.containsKey(recordKey));
+          Assertions.assertEquals(
+              Optional.ofNullable(keys.get(recordKey)).orElse(""),
+              new String(r.getValue(), StandardCharsets.UTF_8));
+          keys.remove(recordKey);
+        }
+        Assertions.assertEquals(0, keys.size());
+      }
+      executorService.shutdown();
+    }
+  }
+
+  private static Stream<? extends Arguments> keyValueFormatArgs() {
+    return Stream.of(
+        Arguments.of(
+            Named.of("Key starting with a single quote",
+                "'key%1$d=>"),
+            Named.of("Value starting with a number ending with a" +
+                " single quote", "%1$dvalue'")
+        ),
+        Arguments.of(
+            Named.of("Key ending with a number", "key%1$d"),
+            Named.of("Value starting & ending with a number", "%1$dvalue%1$d")
+        ),
+        Arguments.of(
+            Named.of("Key starting with a single quote & ending" +
+                " with a number", "'key%1$d"),
+            Named.of("Value starting & ending with a number " +
+                "& elosed within quotes", "%1$d'value%1$d'")),
+        Arguments.of(
+            Named.of("Key starting with a single quote & ending" +
+                " with a number", "'key%1$d"),
+            Named.of("Value starting & ending with a number " +
+                "& elosed within quotes", "%1$d'value%1$d'")
+        ),
+        Arguments.of(
+            Named.of("Key ending with a number", "key%1$d"),
+            Named.of("Value starting & ending with a number " +
+                "& containing null character & new line character",
+                "%1$dvalue\n\0%1$d")
+        ),
+        Arguments.of(
+            Named.of("Key ending with a number & containing" +
+                " a null character", "key\0%1$d"),
+            Named.of("Value starting & ending with a number " +
+                "& elosed within quotes", "%1$dvalue\r%1$d")
+        )
+    );
+  }
+
+  private static byte[] getBytes(Integer val) {
+    byte[] b = new byte[4];
+    for (int i = 3; i >= 0; i--) {
+      b[i] = val.byteValue();
+      val = val >> 8;
+    }
+    return b;
+  }
+
+  private static byte[] getBytes(Long val) {
+    byte[] b = new byte[8];
+    for (int i = 7; i >= 0; i--) {
+      b[i] = (byte)(val & 0xff);
+      val = val >> 8;
+    }
+    return b;
+  }
+
+  private static byte[] getBytes(String val) {
```

Comment:
Use `StringUtils.string2Bytes()` instead of creating new one.

Reply:
no we cannot do that, it will encode into utf-8. I want the raw character byte array, since I am already prepending the bytes of the length in sst dumptool